replication/client/client.go

```diff
@@ -235,19 +235,31 @@ func (c *Replicator) Start(progressChan <-chan uint64) {
 		replicationCtx, cancelFn := context.WithTimeout(c.shutdownHandler.TerminateCtx, 5*time.Second)
 		defer cancelFn()
 
-		message, rplErr = conn.WaitForReplicationMessage(replicationCtx)
+		message, rplErr = pgConn.ReceiveMessage(replicationCtx)
 	}()
 
-	if message == nil || message.ServerHeartbeat == nil {
+
+	cd, ok := message.(*pgproto3.CopyData)
+	if !ok {
+		log.Errorf("Received unexpected message: %#v", message)
+		return
+	}
```

Comment:
So what happens if this returns?

Reply:
Bifrost shuts down. This is a new error case that didn't exist in the old library. I haven't yet added test cases for these yet.